CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+---
+CIP: ?
+Title: Validation zones
+Category: Ledger
+Status: Proposed
+Authors:
+    - Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Implementors: []
+Discussions:
+Created: 2024-03-11
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We propose a set of changes that revolve around validation zones, a construct for allowing greater interdependency between transactions. 
+We use this to enable “unresolved transaction inputs”, and also atomic groups of transactions. 
+We show how these features can be used to address a number of use cases from CPS-????.
+
+## Motivation: why is this CIP necessary?
+
+This CIP provides a partial solution to the problems described in CPS-????. 
+In particular, it describes some ledger changes that allow intent settlement for a wide range of intents that require “counterparty irrelevance”, including most of the swap use cases and dApp fee sponsorship.
+
+## Specification
+
+### Validation zones
+
+A validation zone is a container for transactions. 
+The main property of a validation zone is that the node may need to look at all the transactions in the zone in order to decide if any of them are valid. 
+Within validation zones we can therefore allow features that require “earlier” transactions to depend on “later” transactions for their validity.
+
+At the ledger level, a validation zone:
+
+- Is a new top-level element (an alternative to a transaction)
+- Contains a sequence of transactions (not zones, there is no point nesting them)
+- Carries no further data beyond this
+- Triggers a special mode of validation for the transactions within (see “validating a zone” below)
+
+### Unresolved inputs and resolving outputs
+
+We add a new kind of transaction input, an “unresolved input” (UTXI). 
+An unresolved input differs from a normal transaction input:
```

Comment:
the key property of it is that it looks exactly like an output! It basically will be in the Agda model a specially-marked output

Reply:
I still find the UTXI terminology pretty confusing. I am thinking of this instead :

The UTxO stored in the ledger state will still be `UTxO`, but inside the validation zone, we will instead have `UTxONew` (or with a better name)

`  IsBackwards = Bool
  TxIn     = TxId × Ix
  TxInNew     = TxIn × IsBackwards
  TxOut    = Addr × Value × Maybe DataHash
  TxOutNew = TxOut × IsBackwards
  UTxO     = TxIn ⇀ TxOut
  UTxONew     = TxInNew ⇀ TxOutNew`

Reply:
How about required input / satisfying output? Once I started using that terminology in my head it clicked for me.

Reply:
I talked to Andre at the formal ledger meeting today, and he said that it is always best to put any new features into entirely new transaction fields. So, the transaction will have two new fields, one for UTxIs (Backwards Outputs?), and one for the backwards inputs. I think the UTxIs will have the same type as output (with a different name), and the other things (backwards inputs) will have the same type as inputs. We just need to name these. 

Then we have `UTxOZone = UTxO x BackwardsUTxO` or something like that, and the second component needs to be empty (an empty UTxO map?) for a zone to be valid

Reply:
"requested" and "required" both work i think

Reply:
Yeah, very open to other names. I liked "unresolved" because then we have UTXI which looks like UTXO :grin: A good name will go a long way towards making this easy to understand!

> the key property of it is that it looks exactly like an output! It basically will be in the Agda model a specially-marked output

I discuss this in the doc a bit, but I think we are on the horns of a dilemma:
1. Either we have "outputs" that are weird (negative value) and therefore behave a lot like inputs; or
2. We have "inputs" that are weird (they have spending guards) and therefore behave a lot like outputs.

I think that it's _less_ weird to think that there might be things other than outputs that need spending guards (although it's a new idea!). I also think the cost of the weirdness in case 1 is greater, because it forces all local reasoning about the transaction (e.g. scripts) to deal with negative quantities, which will be a pain for everyone. Case 2 is just a pain for the ledger :joy:

Reply:
I am open to not calling either weird thing without using the input/output terminology, but i hope there is some agreement on the structure of the types, ie. two new fields in Tx, one has an address (possibly only a spending address), value, and datum, and the other has a TxId and an Ix.